example_buffer.py

```diff
@@ -132,10 +150,69 @@ def _rsync_dir(source_dir, dest_dir):
                         stderr=rsync_log)
 
 
-def fill_and_wait(bufsize=dual_net.EXAMPLES_PER_GENERATION,
+def _determine_chunk_to_make(write_dir):
+    models = fsdb.get_models()
+    # Last model is N.  N+1 is training.  We should gather games for N+2.
+    chunk_to_make = os.path.join(write_dir, str(
+        models[-1][0] + 2) + '.tfrecord.zz')
+    while tf.gfile.Exists(chunk_to_make):
+        print("Chunk for next model ({}) already exists.  Sleeping.".format(chunk_to_make))
+        time.sleep(5 * 60)
+    print("Making chunk:", chunk_to_make)
+
+    return chunk_to_make
+
+def fill_and_wait_time(bufsize=dual_net.EXAMPLES_PER_GENERATION,
+                  write_dir=None,
+                  threads=32,
+                  start_from=dt.datetime.utcnow()):
+    write_dir = write_dir or fsdb.golden_chunk_dir()
+    buf = ExampleBuffer(bufsize)
+    chunk_to_make = _determine_chunk_to_make(write_dir)
+
+    hours = _get_hour_dirs()
+    with timer("Rsync"):
+        time_rsync(_earlier_of(dt.datetime.strptime(hours[-1], "%Y-%m-%d-%H/"), start_from))
+        start_from = dt.datetime.utcnow()
+
+    hours = _get_hour_dirs()
+
+    # TODO: this should fast stop after we've got 500k files...
+    files = tqdm(map(lambda d: tf.gfile.Glob(
+                               os.path.join(LOCAL_DIR, d, "*.zz")),
+                     reversed(hours)), total=len(hours))
+    buf.parallel_fill(list(itertools.chain(*files)), threads=threads)
+    print("Filled buffer, watching for new games")
+
+    models = fsdb.get_models()
+    while (fsdb.get_latest_model()[0] == models[-1][0] or
```

Comment:
instead of comparing the first entry of the tuple `x[0] == y[0]`, why not compare the entire thing? Should lead to the same result with less line noise.

Reply:
Done.